contracts/contracts/MACI.sol

```diff
@@ -190,7 +179,7 @@ contract MACI is IMACI, DomainObjs, Params, Utilities, Ownable(msg.sender) {
     address _verifier,
     address _vkRegistry,
     Mode _mode
-  ) public virtual onlyOwner returns (PollContracts memory pollAddr) {
+  ) public virtual returns (PollContracts memory pollAddr) {
```

Comment:
Is it expected that everyone can deploy polls?

Reply:
The only one concern here: poll finalization depends on block timestamp and if someone can manipulate with it, potentially poll can be finished before the deadline.

Reply:
> Is it expected that everyone can deploy polls?

Up to discussion here. I think it could open up scenarios where single MACI instances are re-used by different teams. Imagine Someone uses  MACI for a vote which is gatekept by an event ticket, let's say Devcon. Now imagine each sub-event (or side event) in Devcon wants to launch their own poll (for whatever reason). With one contract only, individual teams can launch completely independent polls (each of them would have a different set of Poll/Mp/Tally contracts) but all of them would be able to re use the already signed up users which are known to have passed the sybil checks at the gatekeeper level.

Reply:
> The only one concern here: poll finalization depends on block timestamp and if someone can manipulate with it, potentially poll can be finished before the deadline.

So you are referring to a block proposer/validator which would try to validate a block with a different block.timestamp to be able to "complete" a poll earlier? I guess eventually there might even be financial incentives to do that, though this would still be a problem (less of a problem of course) with the `onlyOwner` modifier, as it would need to be coordinated between the owner and validator. Perhaps in general relying on block numbers vs block timestamp might be a safer solution. We can keep `onlyOwner` for tree merging for now, and think more about it.

Reply:
is that possible that we could have a list of addresses that could manipulate on the contracts? in the above example (devcon), who need to deploy their own poll would know beforehand, thus they could ask the maci deployer to put them on the whitelist and they are the ones who could deploy polls? limit the risk to trusted addresses.

Reply:
Sure there are access control libraries which you can use to give multiple roles or same role to more people. Though what exactly is the risk of someone deploying a poll? Before we had only owner because you could only have one poll at a time, so if someone deployed a poll before you (contract deployer) could then you'd need a new contract, but now there's not that problem anymore

Reply:
I don't know if there's any risk now. actually I'm just thinking about something maybe unrelated to these discussions, that if we wanna allow anyone to deploy a poll, but the EASGatekeeper still only accept attestation from a certain address, would that be a problem? maybe would be good to keep a list of addresses for EASGatekeeper?

Reply:
> I don't know if there's any risk now. actually I'm just thinking about something maybe unrelated to these discussions, that if we wanna allow anyone to deploy a poll, but the EASGatekeeper still only accept attestation from a certain address, would that be a problem? maybe would be good to keep a list of addresses for EASGatekeeper?

Good questions to bring up, thanks for that. So let's start with the gatekeepers. Any solution should not be focused on a particular gatekeeper like the EAS for instance, so we should talk gatekeepers in general. After all, a Gatekeeper is a contract that given some inputs, either reverts or gives back execution to the MACI contract if they believe the user should be allowed to vote. 

When you say "maybe would be good to keep a list of addresses for EASGatekeeper?" do you mean that we should have more gatekeepers on MACI's side or that an EAS gatekeeper should have a list of trusted attesters vs only one?

Second, just because everyone could deploy a poll from a MACI instance, it doesn't mean that they should nor that they will. They might not know that there's a MACI instance deployed there, or if they do know and they see that the instance uses an EAS gatekeeper, well it's their choice whether they want to run a poll that only users with that attestation can access, or instead they could deploy their own Maci instance. 
In terms of benefit for the original MACI deployers, this comes down to smaller contract due to Ownable not being included, as well as a tiny saving on the `deployPoll` function as there's no `onlyOwner` modifier.

Reply:
> When you say "maybe would be good to keep a list of addresses for EASGatekeeper?" do you mean that we should have more gatekeepers on MACI's side or that an EAS gatekeeper should have a list of trusted attesters vs only one?

I'm thinking about this: *an EAS gatekeeper should have a list of trusted attesters*.
That's why I think this is not quite related to this PR, only related to how we implement EASGatekeeper.

> Second, just because everyone could deploy a poll from a MACI instance, it doesn't mean that they should nor that they will. They might not know that there's a MACI instance deployed there, or if they do know and they see that the instance uses an EAS gatekeeper, well it's their choice whether they want to run a poll that only users with that attestation can access, or instead they could deploy their own Maci instance.

That's quite great to imagine about. Users could deploy their own rounds in a, let's say, certain event, making voting occurs more often. Even though some small things (e.g. what's for dinner? ...) are not quite related to using MACI (?), but it would be interesting that we could do several voting rounds in a group (e.g. devcon attendees, students from class 309, ...) safe and sound.

Reply:
> I'm thinking about this: _an EAS gatekeeper should have a list of trusted attesters_. That's why I think this is not quite related to this PR, only related to how we implement EASGatekeeper.

right yea we can have a EASGatekeeperMultipleAttester contract which has multiple attester. That could be useful but needs to be used with caution.

> That's quite great to imagine about. Users could deploy their own rounds in a, let's say, certain event, making voting occurs more often. Even though some small things (e.g. what's for dinner? ...) are not quite related to using MACI (?), but it would be interesting that we could do several voting rounds in a group (e.g. devcon attendees, students from class 309, ...) safe and sound.

Yea totally a good use case, which is possible now but only with 1 gatekeeper at a time (for now)